Add io.github.alexays.Waybar

hholst80 opened the pull request:
Highly customizable Wayland bar for Sway and
Wlroots based compositors.

Upstream: https://github.com/alexays/waybar

<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [X] Highly customizable Wayland bar for Sway and Wlroots based compositors.
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I contacted upstream developers about this submission. https://github.com/Alexays/Waybar/issues/3671

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,262 @@
+app-id: io.github.alexays.waybar
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
```
current runtime is 24.08

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,262 @@
+command: waybar
+finish-args:
+  - --socket=wayland
+  # NOTE: To make the linter happy: Waybar obviously expects Wayland.
+  - --socket=fallback-x11
```
no. you need to apply for an exception.

hholst80 replied:
Done. https://github.com/flathub-infra/flatpak-builder-lint/pull/596

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,262 @@
+  - --share=ipc
+  - --device=dri
+  - --socket=pulseaudio
+  # DBus access for system and session (for tray, power, etc.)
+  # - --socket=session-bus
+  # - --socket=system-bus
+  # Allow reading host-installed icons (for tray icons)
+  # - --filesystem=/var/lib/flatpak:ro
+  # - --filesystem=xdg-data/flatpak:ro
+  # Environment to include Flatpak-hosted icons
+  # - --env=XDG_DATA_DIRS=/var/lib/flatpak/exports/share:/app/share:/usr/share:/usr
```
```suggestion
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,262 @@
+  # - --env=XDG_DATA_DIRS=/var/lib/flatpak/exports/share:/app/share:/usr/share:/usr
+
+modules:
+  - name: atk
```
this is in the runtime

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,262 @@
+        url: https://download.gnome.org/sources/atk/2.38/atk-2.38.0.tar.xz
+        sha256: ac4de2a4ef4bd5665052952fe169657e65e895c5057dffb3c2a810f6191a0c36
+
+  - name: gdk-pixbuf
```
this is in the runtime

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,262 @@
+        url: https://download.gnome.org/sources/gdk-pixbuf/2.42/gdk-pixbuf-2.42.12.tar.xz
+        sha256: b9505b3445b9a7e48ced34760c3bcb73e966df3ac94c95a148cb669ab748e3c7
+
+  - name: pango
```
this is in the runtime

hfiguiere commented on flathub.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
why?

hholst80 replied:
It's the only thing I can test right now.

hfiguiere replied:
clearly that won't change if you keep it that way.

hholst80 replied:
What do you suggest?

hfiguiere replied:
obviously, to just remove it

hfiguiere commented on flathub.json:
just remove the file

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  # TODO: This build is still very minimal and more components should to be enabled and tested.
+  
+  # - name: intltool
+  #   buildsystem: autotools
+  #   sources:
+  #     - type: archive
+  #       url: https://launchpad.net/intltool/trunk/0.51.0/+download/intltool-0.51.0.tar.gz
+  #       sha256: 67c74d94196b153b774ab9f89b2fa6c6ba79352407037c8c14d5aeb334e959cd
+
+  # - name: libdbusmenu
```
if you even build these two use the shared-modules. So you can remove all this commented part.

hholst80 replied:
is there a libdbusmenu in shared? I didn't find it, is it part of another package?

hfiguiere replied:
my bad, this one isn't.

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+modules:
+  - name: libsigc++
+    buildsystem: meson
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: glibmm
+    buildsystem: meson
+    build-options:
+      config-opts:
+        - -Dbuild-examples=false
```
```suggestion
    config-opts:
        - -Dbuild-examples=false
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: gtkmm
+    buildsystem: meson
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: chrono-date
+    buildsystem: cmake-ninja
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: fmt
+    buildsystem: cmake-ninja
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+    buildsystem: cmake-ninja
+    build-options:
+      config-opts:
+        - -DCMAKE_BUILD_TYPE=None
```
is `None` really a valid type? Usually this is `Release` or `RelWithDebInfo`

hholst80 replied:
Release sounds much more reasonable.

hholst80 replied:
I added release build to the meson packages too. (Unlikely to have any impact on an app like Waybay...)

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: spdlog
+    buildsystem: cmake-ninja
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: jsoncpp
+    buildsystem: meson
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,202 @@
+
+  - name: waybar
+    buildsystem: meson
+    build-options:
+      config-opts:
```
```suggestion
    config-opts:
```

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,235 @@
+        url: https://github.com/wmww/gtk-layer-shell/archive/v0.9.0.tar.gz
+        sha256: 3809e5565d9ed02e44bb73787ff218523e8760fef65830afe60ea7322e22da1c
+
+  - name: intltool
```
as I mentionned earlier, use the shared-modules for `intltool`.

hholst80 replied:
Sorry missed it.

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,235 @@
+        url: https://launchpad.net/intltool/trunk/0.51.0/+download/intltool-0.51.0.tar.gz
+        sha256: 67c74d94196b153b774ab9f89b2fa6c6ba79352407037c8c14d5aeb334e959cd
+
+  - name: lua
```
and also for lua

hfiguiere commented on io.github.alexays.waybar.yaml:
```diff
@@ -0,0 +1,235 @@
+        - -Dsystemd=disabled
+        - -Dman-pages=disabled
+        - -Dexperimental=false
+        - -Djack=disabled
```
`pw-jack` is in the runtime too.

hholst80 commented:
I have now addressed all the comments raised by @hfiguiere. Thank you so much for your swift feedback and also making the package much better!

I kind of feel now that we should have libdbusmenu part of the initial release as well. Let me look at that tomorrow Monday and see if I can get that working. If not, not a huge deal, but it would be nice to have at least all the features I use myself in the first release of this package.

hholst80 commented:
Got it working. Looks terrible though, the things one has to do to compile 10 year old packages... It's not pretty. Seeing my old telegram task tray again makes it all worth it.

![image](https://github.com/user-attachments/assets/97398de9-0134-4550-8fb1-f96ca6766fcf)

bbhtt commented:
A wayland bar does not qualify as an "app".

bbhtt commented:
Unfortunately no (it's written in the submission policy section of requirements that we prefer desktop apps). 

Flatpak isn't very well suited for these kind of components, you can make it work but it's easier to handle this with other packaging systems. It'd mostly require some kind of autostarting either via WM or some autostart daemon which Flatpak cannot do ootb and then these are all highly customisable through config files, having the config in a different location (`~/.var/app/$appid/config`) is going to be an issue.

hholst80 commented:
Honestly for me this is exactly a desktop app. The only difference is that it picks a very opinionated, but configurable, place at the screen.

For now, I'll self-host it, and we can revisit this if the trends are changing. I think users in the nisch 'highly customized desktop power users' would appreciate being able to run exactly the right versioned application independent of the underlying OS. Waybar stands out as one of the most powerful bars, but it is also quite a mess to compile from source.  Users are therefore practically limited to whatever version that is available in the distro. 

As to practical problems to integrating waybar I do not think there is much problem in the cases where anyone would like to use waybar.  These window managers will have a shell script like syntax to start a bar (or any other general application) and it won't matter if that is a flatpak or not.
